DEVFILE_REGISTRY.md

```diff
@@ -61,3 +61,171 @@ If you want to send telemetry information to your own Segment instance, specify
+The container image used by default is `quay.io/devfile/registry-viewer:next` and you can change it with the field `spec.registryViewer.image`.
+
+```bash
+$ cat <<EOF | oc apply -f -
+apiVersion: registry.devfile.io/v1alpha1
+kind: DevfileRegistry
+metadata:
+  name: my-devfile-registry
+spec:
+  devfileIndex:
+    image: quay.io/my-devfile/devfile-index:next
+  ociRegistry:
+    image: quay.io/my-devfile/oci-registry:next
+  registryViewer:
+    image: quay.io/my-devfile/registry-viewer:next
+EOF
+```
+
+### Defining the ImagePullPolicy for pulling containers
+
+By default, the containers will be pulled depending on the policy set on the Kubernetes or OpenShift cluster the registry is deployed on.
+
+You can set a specific pull policy for each container, by setting the fields `spec.devfileIndex.imagePullPolicy`, `spec.ociRegistry.imagePullPolicy` and `spec.registryViewer.imagePullPolicy` to a value `IfNotPresent`, `Always` or `Never`. See the [Kubernetes documentation](https://kubernetes.io/docs/concepts/containers/images/#image-pull-policy) for more information.
+
+
+```bash
+$ cat <<EOF | oc apply -f -
+apiVersion: registry.devfile.io/v1alpha1
+kind: DevfileRegistry
+metadata:
+  name: my-devfile-registry
+spec:
+  devfileIndex:
+    image: quay.io/my-devfile/devfile-index:next
+    imagePullPolicy: Always
+  ociRegistry:
+    image: quay.io/my-devfile/oci-registry:next
+    imagePullPolicy: Always
+  registryViewer:
+    image: quay.io/my-devfile/registry-viewer:next
+    imagePullPolicy: Always
+EOF
+```
+
+## Disabling the web frontend 
+
+You can ask the operator to deploy the Devfile Registry without the `registry-viewer` container, by setting the field `spec.headless` to `true`.
+
+```bash
+$ cat <<EOF | oc apply -f -
+apiVersion: registry.devfile.io/v1alpha1
+kind: DevfileRegistry
+metadata:
+  name: headless-devfile-registry
+spec:
+  devfileIndex:
+    image: quay.io/devfile/devfile-index:next
+  telemetry:
+    registryName: test
+  headless: true
+EOF
+```
+
+## Using a Persistent Volume to store OCI images
+
+The Devfile stacks are stored in OCI images served by the OCI registry container. By default, an Ephemeral volume is created and used to store these images.
+
+You can ask the operator to create a Persistent Volume and to mount it into the `oci-registry` container's filesystem, so the OCI images are stored into this persistent volume, by setting the field `spec.storage.enabled` to `true`.
+
+By default, the requested size for the Persistent Volume will be `1Gi`. You can ask to set a specific size with the field `spec.storage.ociRegistryImage`
+
+```bash
+$ cat <<EOF | oc apply -f -
+apiVersion: registry.devfile.io/v1alpha1
+kind: DevfileRegistry
+metadata:
+  name: devfile-registry
+spec:
+  devfileIndex:
+    image: quay.io/devfile/devfile-index:next
+  telemetry:
+    registryName: test
+  storage:
+    enabled: true
+    ociRegistryImage: 2Gi
```

Comment:
This should be `registryVolumeSize` IMO, but the code currently indicate this:

```
	RegistryVolumeSize string `json:"ociRegistryImage,omitempty"`

```

Do we want to change the code directly, or do we want to deprecate it?

Reply:
Preferably we should fix it before documenting it as it's a typo.